lib/modules/manager/npm/post-update/npm.spec.ts

```diff
@@ -162,6 +160,54 @@ describe('modules/manager/npm/post-update/npm', () => {
     expect(execSnapshots).toEqual([]);
   });
 
+  it('deduplicates dependencies on installation with NPM >= 7', async () => {
+    const execSnapshots = mockExecAll();
+    // package.json
+    fs.readLocalFile.mockResolvedValueOnce('{}');
+    fs.readLocalFile.mockResolvedValueOnce('package-lock-contents');
+    const postUpdateOptions = ['npmDedupe'];
+    const updates = [
+      { packageName: 'some-dep', newVersion: '1.0.1', isLockfileUpdate: false },
+    ];
+    const res = await npmHelper.generateLockFile(
+      'some-dir',
+      {},
+      'package-lock.json',
+      { postUpdateOptions },
+      updates
+    );
+    expect(fs.readLocalFile).toHaveBeenCalledTimes(2);
+    expect(res.error).toBeFalse();
+    expect(res.lockFile).toBe('package-lock-contents');
+    expect(execSnapshots).toHaveLength(1);
+    expect(execSnapshots[0].cmd).toContain('--prefer-dedupe');
+    expect(execSnapshots[0].cmd).toContain('--package-lock-only');
```

Comment:
prefer to use `toMatchObject`

Reply:
@rarkins asked me to refactor away from using snapshots. (https://github.com/renovatebot/renovate/pull/25466#discussion_r1375258361) Wouldn't `expect(execSnapshot).toMatchObject()` be the same as a snaphot inlined into the test file itself? It's unclear to me how new tests should look like to be honest. Is there any documentation or best practice examples available?

Reply:
The [Development best practices docs](https://github.com/renovatebot/renovate/blob/main/docs/development/best-practices.md) has a section about unit testing, and other tips.